src/algebraic_geometry/morphisms/finite_type.lean

```diff
@@ -0,0 +1,100 @@
+/-
+Copyright (c) 2022 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import algebraic_geometry.morphisms.ring_hom_properties
+import algebraic_geometry.morphisms.quasi_compact
+import ring_theory.ring_hom.finite_type
+
+/-!
+# Morphisms of finite type
+
+A morphism of schemes `f : X ⟶ Y` is locally of finite type if for each affine `U ⊆ Y` and
+`V ⊆ f ⁻¹' U`, The induced map `Γ(Y, U) ⟶ Γ(X, V)` is of finite type.
+
+A morphism of schemes is of finite type if it is both locally of finite type and quasi-compact.
+
+We show that these properties is local, and is stable under compositions.
+
+-/
+
+noncomputable theory
+
+open category_theory category_theory.limits opposite topological_space
+
+universe u
+
+namespace algebraic_geometry
+
+variables {X Y : Scheme.{u}} (f : X ⟶ Y)
+
+/--
+A morphism is `affine` if the preimages of affine open sets are affine.
+-/
+@[mk_iff]
+class locally_of_finite_type (f : X ⟶ Y) : Prop :=
+(finite_type_of_affine_subset :
+  ∀ (U : Y.affine_opens) (V : X.affine_opens) (e : V.1 ≤ (opens.map f.1.base).obj U.1),
+  (f.1.c.app (op U) ≫ X.presheaf.map (hom_of_le e).op).finite_type)
+
+lemma locally_of_finite_type_eq :
+  @locally_of_finite_type = affine_locally @ring_hom.finite_type :=
```

Comment:
Is there a reason why you don't take this as the definition?

Reply:
One reason is that I would like `locally_of_finite_type` to be a class. Another is that `affine_locally` gives bad definitional equalities.